Pinta.Core/Effects/ColorDifference.cs

```diff
@@ -11,78 +11,75 @@
 namespace Pinta.Core;
 
 /// <summary>
-/// ColorDifferenctEffect is a base class for my difference effects
+/// ColorDifferenctEffect is a utility class for difference effects
 /// that have floating point (double) convolution filters.
 /// its architecture is just like ConvolutionFilterEffect, adding a
-/// function (RenderColorDifferenceEffect) called from Render in each
-/// derived class.
+/// function (RenderColorDifferenceEffect) called from Render.
 /// It is also limited to 3x3 kernels.
 /// (Chris Crosetto)
 /// </summary>
-public abstract class ColorDifferenceEffect : BaseEffect
+public static class ColorDifference
 {
 	public static void RenderColorDifferenceEffect (
 		double[,] weights,
-		ImageSurface src,
-		ImageSurface dest,
+		ImageSurface source,
+		ImageSurface destination,
 		ReadOnlySpan<RectangleI> rois)
 	{
-		RectangleI src_rect = src.GetBounds ();
+		if (weights.GetLength (0) != 3 || weights.GetLength (1) != 3) throw new ArgumentException ("Must be a 3x3 array", nameof (weights));
 
-		// Cache these for a massive performance boost
-		var src_data = src.GetReadOnlyPixelData ();
-		var dst_data = dest.GetPixelData ();
-		int src_width = src.Width;
+		RectangleI surfaceBounds = source.GetBounds ();
 
-		foreach (RectangleI rect in rois) {
-
-			foreach (var pixel in Utility.GeneratePixelOffsets (rect, src.GetSize ())) {
+		ReadOnlySpan<ColorBgra> sourceData = source.GetReadOnlyPixelData ();
+		Span<ColorBgra> destinationDataa = destination.GetPixelData ();
```

Comment:
There's a typo here: `Dataa`

Reply:
@cameronwhite thanks for pointing it out. I just fixed it